src/sas/qtgui/Utilities/Preferences/PreferencesPanel.py

```diff
@@ -94,6 +100,16 @@ def _saveStagedChanges(self):
         """ When OK or Apply are clicked, all staged changes should be applied to the config. """
         for k, v in self._staged_changes.items():
             setattr(config, k, v)
+        if any(self._staged_requiring_restart):
+            message = "SasView must restart for the following values to take effect. Do you wish to restart?:\n"
+            for val in self._staged_requiring_restart:
+                message += f"\t-{val}\n"
+            msgBox = QMessageBox(QMessageBox.Information, "", message, QMessageBox.Yes|QMessageBox.No)
+            msgBox.show()
+            if msgBox.exec() == QMessageBox.Yes:
+                self.parent.guiManager.quitApplication()
+                os.execl(sys.executable, os.path.abspath(__file__), *sys.argv)
```

Comment:
Interestingly, this works fine in the installed version, despite `__file__` pointing to relative location of sasview.exe in pyinstaller generated setups. 
If you open the python shell in the installed version, the `__file__` attribute is not even defined.
I would probably use `sys.executable` instead of `os.path.abspath(__file__)`, since in the installed version it points to location of the exe file (without the need to expand the path with `abspath`) and in the dev build, it points to python.exe, with sys.argv containing `run.py` and other arguments.

But since this is working as is, I wouldn't put too much importance in the abve.